modules/reference/pages/data-transform-rust-sdk.adoc

```diff
@@ -0,0 +1,130 @@
+= Rust SDK for Data Transforms
+:description: Work with data transforms using Rust.
+
+Rust SDK reference documentation for Redpanda data transforms.
+
+== Data transforms
+
+The `redpanda_transform_sdk` crate contains the core functions and types for transforming data within Redpanda.
+
+=== Data transforms functions
+
+==== on_record_written
+
+[,rust,role="no-copy"]
+----
+pub fn on_record_written<E, F>(cb: F) -> !
+where
+    E: Debug,
+    F: Fn(WriteEvent<'_>, &mut dyn RecordWriter) -> Result<(), E>,
+----
+
+The `on_record_written` function transforms individual records after they have been written to an input topic. The function registers a callback to be fired when a record is written to the topic. This callback is triggered after the record has been written and fsynced to disk and the producer has been acknowledged.
+
+Since this method blocks and runs forever, it should be called from the `main` function. Perform the necessary steps prior to transforming any records before calling this method.
+
+---
+
+=== Data transforms types
+
+==== Structs
+
+===== BorrowedHeader
+
+[,rust]
+----
+pub struct BorrowedHeader<'a> { /* private fields */ }
+----
+
+`BorrowedHeader` is a zero-copy <<borrowedrecord,BorrowedRecord>> header.
+
+===== BorrowedRecord
+
+[,rust]
+----
+pub struct BorrowedRecord<'a> { /* private fields */ }
+----
+
+`BorrowedRecord` is a zero-copy representation of a <<record,Record>> in Redpanda. 
+
+===== Record
+
+[,rust]
+----
+pub struct Record { /* private fields */ }
+----
+
+`Record` is a record in Redpanda. It consists of a key-value pair of bytes, along with a collection of <<recordheader,RecordHeader>>.
+
+Records are generated as the result of any transforms that act upon a `BorrowedRecord`.
+
+===== RecordHeader
+
+[,rust]
+----
+pub struct RecordHeader { /* private fields */ }
+----
+
+A `RecordHeader` is a key-value pair attached to a <<record,Record>>. Headers are opaque to the broker and are purely a mechanism for the producer and consumers to pass information.
+
+===== RecordWriter
+
+[,rust]
+----
+pub struct RecordWriter<'a> { /* private fields */ }
+----
+
+`RecordWriter` is a struct that writes transformed records to the output topic.
+
+===== WriteEvent
+
+[,rust]
+----
+pub struct WriteEvent<'a> {
+    pub record: WrittenRecord<'a>,
+}
+----
+
+`WriteEvent` is an event generated after a write event within the broker. These events are asynchronously triggered after the broker acknowledges the producer's write request.
+
+===== WrittenRecord
+
+[,rust]
+----
+pub struct WrittenRecord<'a> { /* private fields */ }
+----
+
+A written <<record,Record>> within Redpanda. A `WrittenRecord` is handed to <<on_record_written,on_record_written>> event handlers as the record that Redpanda wrote. The record contains a key-value pair with some headers and the record's timestamp.
+
+==== Enums
```

Comment:
maybe a short intro to describe enums with a link to the Rust doc: https://doc.rust-lang.org/rust-by-example/custom_types/enum.html